pos.lua

```diff
@@ -6,8 +6,6 @@ local S = minetest.get_translator("areas")
 -- Since this is mostly copied from WorldEdit it is mostly
 -- licensed under the AGPL. (select_area is a exception)
 
-areas.marker1 = {}
-areas.marker2 = {}
```

Comment:
These two could simply be made local. You don't need the complicated logic below.
```Lua
local marker1 = {}
local marker2 = {}
```

Reply:
Saving entities is discouraged, so I wrote my code. If you decide to keep saving g entries, there would still be something to improve, and I am happy to help with.

Reply:
True. In 5.10.0-dev, further issues in that regard were fixed. The engine needs to properly deal with all situations anyway. An ID-based approach to check the existence of the entity could be an alternative, but I am currently not aware of any such method that would be available and documented by the engine.
Your code makes the best out of this situation.